languages/java/oso/src/main/java/com/osohq/oso/Ffi.java

```diff
@@ -35,17 +38,22 @@ protected int loadStr(String src, String filename) throws Exceptions.OsoExceptio
         }
 
         protected Query newQueryFromStr(String queryStr) throws Exceptions.OsoException {
-            return new Query(checkResult(polarLib.polar_new_query(ptr, queryStr, 0)));
+            Pointer query = checkResult(polarLib.polar_new_query(ptr, queryStr, 0));
+            processMessages();
```

Comment:
It's not a huge deal, but seems annoying to have to add processMessages everywhere. Is there an alternate point we could add it? (For example, in the checkResult method?).

Reply:
ordering